pkgs/development/compilers/gerbil/gerbil-support.nix

```diff
@@ -1,53 +1,112 @@
-{ pkgs, lib, gccStdenv, callPackage, fetchFromGitHub }:
-# See ../gambit/build.nix regarding gccStdenv
-
-rec {
-  # Gerbil libraries
-  gerbilPackages-unstable = {
-    gerbil-libp2p = callPackage ./gerbil-libp2p.nix { };
-    gerbil-utils = callPackage ./gerbil-utils.nix { };
-    gerbil-crypto = callPackage ./gerbil-crypto.nix { };
-    gerbil-poo = callPackage ./gerbil-poo.nix { };
-    gerbil-persist = callPackage ./gerbil-persist.nix { };
-    gerbil-ethereum = callPackage ./gerbil-ethereum.nix { };
-    smug-gerbil = callPackage ./smug-gerbil.nix { };
-  };
+{ pkgs, lib, callPackage }:
+
+let POP = lib.POP; in
+with POP;
+
+selfPop (gerbil-support: with gerbil-support; {
+  inherit POP;
+
+  prePackages-unstable =
+    let names = ["gerbil-libp2p" "smug-gerbil"
+                 "gerbil-utils" "gerbil-crypto" "gerbil-poo" "gerbil-persist" "gerbil-ethereum"
+                 "glow-lang"];
+        pk = name: callPackage "${./.}/${name}.nix" (unpop prePackage-defaults);
```

Comment:
This ends up being an import from store, which is not allowed in restricted settings (Hydra, ofBorg, etc.) and thus also not used in Nixpkgs

Reply:
I don't understand at all what you mean. What is an "import from store"?

Reply:
When the string you pass to `callPackage` gets constructed, `${./.}` gets expanded by copying the current directory to the store. So you get an absolute path pointing into store. Except that read-only evaluation computes the target path but does not copy things into the store until the evaluation is done, so you are not allowed to import from there during the evaluation.

Reply:
OK, so how am I supposed to import from `./foo.nix` ?

Reply:
OK, I found a workaround: go from path to name instead of the other way around.